### ISSUE TITLE
Remove parallel transfer over high ports

### ISSUE THREAD

**Issue by korydraughn** [irods/irods#7950]
Removal of parallel transfer over the high ports implies that the icommands must support parallel transfer over port 1247.

We'll also need to consider how this affects federation between iRODS 4.x and iRODS 5.0.

Requires resolution of #7949 first.

**Comment by alanking** [comment]
Just making a note that server-to-server transfers (copy, replication, phymv, etc.) also use high ports for parallel transfer.

For federation, copying from a 4.x server to a 5.x server would be affected if legacy parallel transfer is removed.

**Comment by trel** [comment]
4.3 could learn to use multi-1247 internally...

so we would still be federation compatible with 4.x (whatever version added multi-1247).

it's not clear whether that satisfies our promise.

**Comment by trel** [comment]
our original promise was only that 4 would remain compatible with 3.

**Comment by alanking** [comment]
We will have to maintain both implementations in 4.3 in order to remain compatible with 3.x, switching based on the version of the remote zone server to which we are connected.

Additionally, replacing all of the internal transfer machinery in a minor release is high risk. Perhaps this can be an opt-in configuration, like we had discussed with replacing the GenQuery1 parser? https://github.com/irods/irods/pull/7819#issuecomment-2179338356

**Comment by trel** [comment]
Agreed on the swapping being high risk.   Feels like default use of multi-1247 internally should be a 5.0, itself.

**Comment by korydraughn** [comment]
> our original promise was only that 4 would remain compatible with 3.

You're saying no claims have been about federation between v5 and v4, correct?

> Additionally, replacing all of the internal transfer machinery in a minor release is high risk. Perhaps this can be an opt-in configuration, like we had discussed with replacing the GenQuery1 parser? [#7819 (comment)](https://github.com/irods/irods/pull/7819#issuecomment-2179338356)

An opt-in seems the safest route.

Alternatively, v5 doesn't federate with v4 at all?

I'm sure that's not a popular idea, but it's worth talking about to determine what that would mean.

**Comment by trel** [comment]
> You're saying no claims have been about federation between v5 and v4, correct?

Correct.

A slightly different conversation is about 4.x clients talking to 5 servers.   Seems we have to support 'both' parallel mechanisms for a period of time (5.x may have default multi-1247, but still support high ports) with 6.x dropping the high ports completely.
